GitTfs.VsCommon/TfsHelper.Common.cs

```diff
@@ -812,7 +812,13 @@ public ITfsChangeset GetShelvesetData(IGitTfsRemote remote, string shelvesetOwne
             var shelveset = shelvesets.First();
 
             var itemSpec = new ItemSpec(remote.TfsRepositoryPath, RecursionType.Full);
-            var change = VersionControl.QueryShelvedChanges(shelveset, new ItemSpec[] { itemSpec }).Single();
+            var changes = VersionControl.QueryShelvedChanges(shelveset, new ItemSpec[] {itemSpec});
+            if (!changes.Any())
+            {
+                throw new GitTfsException("There is no changes in this shelveset that apply to the current tfs remote.")
+                    .WithRecommendation("Try to apply the shelveset on another remote.");
+            }
+            var change = changes.Single();
```

Comment:
:+1:

Another slightly shorter option would be to switch the `Single()` to `SingleOrDefault()` and do a null check.

```csharp
var change = VersionControl.QueryShelvedChanges(shelveset, new ItemSpec[] { itemSpec }).SingleOrDefault();
if (change == null) {
  throw ...
}
```

I like the brevity of `SingleOrDefault`, but either is fine.